bump_version.sh

```diff
@@ -9,46 +15,68 @@ set -o pipefail
 VERSION_FILE=src/version.txt
 README_FILE=README.md
 
-HELP_INFORMATION="bump_version.sh (show|major|minor|patch|prerelease|build|finalize)"
+function usage {
+  cat << HELP
+Usage:
+  ${0##*/} (show|major|minor|patch|finalize)
+  ${0##*/} (build|prerelease) [token]
 
-old_version=$(sed -n "s/^__version__ = \"\(.*\)\"$/\1/p" $VERSION_FILE)
-# Comment out periods so they are interpreted as periods and don't
-# just match any character
-old_version_regex=${old_version//\./\\\.}
+Notes:
+  - If you specify a token it will only be used if the current version is
+    tokenless or if the provided token matches the token used in the current
+    version.
+HELP
+  exit 1
+}
 
-if [ $# -ne 1 ]; then
-  echo "$HELP_INFORMATION"
+function update_version {
+  # Comment out periods so they are interpreted as periods and don't
+  # just match any character
+  old_version_regex=${1//\./\\\.}
+
+  echo Changing version from "$1" to "$2"
+  tmp_file=/tmp/version.$$
+  sed "s/$old_version_regex/$2/" $VERSION_FILE > $tmp_file
+  mv $tmp_file $VERSION_FILE
+  sed "s/$old_version_regex/$2/" $README_FILE > $tmp_file
+  mv $tmp_file $README_FILE
+  git add $VERSION_FILE $README_FILE
+  git commit --message "$3"
+  git push
+}
+
+if [ $# -lt 1 ] || [ $# -gt 2 ]; then
+  usage
 else
+  old_version=$(sed -n "s/^__version__ = \"\(.*\)\"$/\1/p" $VERSION_FILE)
   case $1 in
-    major | minor | patch | prerelease | build)
+    major | minor | patch)
+      if [ $# -ne 1 ]; then
+        usage
+      fi
       new_version=$(python -c "import semver; print(semver.bump_$1('$old_version'))")
-      echo Changing version from "$old_version" to "$new_version"
-      tmp_file=/tmp/version.$$
-      sed "s/$old_version_regex/$new_version/" $VERSION_FILE > $tmp_file
-      mv $tmp_file $VERSION_FILE
-      sed "s/$old_version_regex/$new_version/" $README_FILE > $tmp_file
-      mv $tmp_file $README_FILE
-      git add $VERSION_FILE $README_FILE
-      git commit -m"Bump version from $old_version to $new_version"
-      git push
+      update_version "$old_version" "$new_version" "Bump version from $old_version to $new_version"
+      ;;
+    build | prerelease)
+      if [ $# -eq 2 ]; then
+        new_version=$(python -c "import semver; print(semver.bump_$1('$old_version', token='$2'))")
```

Comment:
If the provided token doesn't match the existing token, it looks `semver` doesn't modify the version or throw an error:
```
>>> import semver
>>> version_info = semver.VersionInfo.parse("1.2.3-rc.4")
>>> version_info.bump_prerelease()
Version(major=1, minor=2, patch=3, prerelease='rc.5', build=None)
>>> version_info.bump_prerelease(token="foo")
Version(major=1, minor=2, patch=3, prerelease='rc.5', build=None)
```

Is that the desired behavior here?  Can you please share output of a manual run of this code in this case so that we can see how the script behaves?

Reply:
You are not mirroring how this is done in the script in your example. The note in the help section clarifies that the provided token is not used if it does not match, but it will still version bump:

```console
>>> import semver
>>> old_version = "1.2.3-rc.4"
>>> semver.bump_prerelease(old_version)
'1.2.3-rc.5'
>>> semver.bump_prerelease(old_version, token="test")
'1.2.3-rc.5'
>>> old_version = "1.2.3"
>>> semver.bump_prerelease(old_version)
'1.2.3-rc.1'
>>> semver.bump_prerelease(old_version, token="test")
'1.2.3-test.1'
```

Reply:
Alright, I'm tracking now.  I was a little confused before, but this all makes sense.

I don't totally love the silent no-op if you attempt to use a different token than the current version, but your note in the usage text does indeed mention this behavior.

Reply:
Yes I'm not a fan of it silently ignoring the provided token, but the Python logic completely ignores the token if the version string is already a prerelease per the [sourec code](https://github.com/python-semver/python-semver/blob/8fe4ef678e1cbacd904fbd9f67a8cb3f971642b0/src/semver/version.py#L326-L333). Making sure to explicitly mention/document the behavior was the only thing I could see doing since I'd rather expose the functionality even with the caveat.